src/lua_test.c

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright (C) 2024 Kentaro Hayashi
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ * Authors:
+ *   Kentaro Hayashi <kenhys at xdump.org>
+ **/
+
+#include "collectd.h"
+#include "lua.c"
+#include "testing.h"
+#include "utils_lua.c"
+
+static lua_State *L = NULL;
+static char config_path[PATH_MAX] = {0};
+const char *module_key_value_config = "<Plugin lua>\n"
+                                      "  Script \"example.lua\"\n"
+                                      "  <Module>\n"
+                                      "    Key \"Value\"\n"
+                                      "  </Module>\n"
+                                      "</Plugin>\n";
+
+void test_setup_config(const char *config) {
+  int fd = 0;
+  FILE *fp = NULL;
+  {
+    /* create temporary .conf */
+    strncpy(config_path, "lua-XXXXXX.conf", strlen("lua-XXXXXX.conf") + 1);
+    fd = mkstemps(config_path, /* .conf */ 5);
+    fp = fdopen(fd, "w+");
+    fprintf(fp, "%s", config);
+    fclose(fp);
+  }
+  {
+    /* create empty Lua script which is loaded via lua_config */
+    fp = fopen("example.lua", "w+");
+    fclose(fp);
+  }
+}
+
+void test_setup(void) {
+  L = luaL_newstate();
+  luaL_openlibs(L);
+}
+
+void test_teardown(void) { lua_close(L); }
+
+DEF_TEST(luaC_pushoconfigitem) {
```

Comment:
This is a great test, thank you!

```suggestion
DEF_TEST(lua_config) {
```

Reply:
Thanks, fixed.